### PR TITLE
Adds simple CSW harvesting

### PR DESCRIPTION
**Pull request by benjwadams**

### CODE REVIEW

File: catalog_harvesting/harvest.py [hunk 1]
```diff
@@ -130,13 +137,79 @@ def download_waf(db, harvest, src, dest):
...
+                rec["validation_errors"].append({
+                    "line_number": "?",
+                    "error": "Invalid Geometry. See gmd:EX_GeographicBoundingBox"
+                })
+            rec['csw_endpoint'] = csw_url
+            rec['update_time'] = datetime.now()
+            rec['harvest_id'] = harvest['_id']
+            if len(rec['validation_errors']):
+                errors += 1
+            count += 1
```
Comment (lukecampbell) [inline catalog_harvesting/harvest.py]: These nested trys inside a for-loop needs to be refactored into separate functions.

File: catalog_harvesting/harvest.py [hunk 2]
```diff
@@ -130,13 +137,79 @@ def download_waf(db, harvest, src, dest):
...
+                    "error": "XML Syntax Error: %s" % e.message
+                }]
+            }
+            errors += 1
+            count += 1
+            get_logger().error(err_msg)
+        except:
+            get_logger().exception("Failed to create record: %s", link)
+            raise
+        print(rec)
```
Comment (lukecampbell) [inline catalog_harvesting/harvest.py]: Please use logger over prints

**Review by lukecampbell — CHANGES_REQUESTED** [review]

File: catalog_harvesting/harvest.py (hunk 2 above)
Comment (benjwadams) [reply catalog_harvesting/harvest.py]: I'm just going to remove that entirely, was for debug and forgot to remove it.

File: catalog_harvesting/harvest.py [hunk 3]
```diff
@@ -158,55 +161,63 @@ def download_csw(db, harvest, csw_url, dest):
...
+        csw.getrecords2(outputschema=namespaces['gmd'], #Return ISO 19115 metadata
+                        startposition=rec_offset,
+                        esn='full', maxrecords=100)
+
+        for name, raw_rec in csw.records.items():
+            # replace slashes with underscore so writing to file does not
+            # cause missing file
+            name_sanitize = name.replace('/', '_')
+            file_loc = os.path.join(dest, name_sanitize + '.xml')
+            print(file_loc)
```
Comment (lukecampbell) [inline catalog_harvesting/harvest.py]: use logging please

Comment (lukecampbell) [inline catalog_harvesting/harvest.py]: Please use logging

**Review by lukecampbell — CHANGES_REQUESTED** [review]

**Review by lukecampbell — APPROVED** [review]

**Pull request merged**